lmcache/experimental/storage_backend/storage_manager.py

```diff
@@ -0,0 +1,250 @@
+import threading
+from concurrent.futures import Future, ThreadPoolExecutor
+from typing import Dict, List, Optional, Tuple
+
+import torch
+from sortedcontainers import SortedDict
+
+from lmcache.config import LMCacheEngineMetadata
+from lmcache.experimental.config import LMCacheEngineConfig
+from lmcache.experimental.memory_management import (BufferMemoryObj,
+                                                    MemoryAllocatorInterface,
+                                                    MemoryFormat, MemoryObj)
+from lmcache.experimental.storage_backend import CreateStorageBackends
+from lmcache.experimental.storage_backend.abstract_backend import \
+    StorageBackendInterface
+from lmcache.logging import init_logger
+from lmcache.utils import CacheEngineKey
+
+logger = init_logger(__name__)
+
+
+# TODO: extend this class to implement caching policies and eviction policies
+class StorageManager:
+    """
+    The StorageManager is responsible for managing the storage backends.
+    """
+
+    def __init__(self, config: LMCacheEngineConfig,
+                 metadata: LMCacheEngineMetadata,
+                 allocator: MemoryAllocatorInterface):
+        self.memory_allocator = allocator
+        self.hot_cache = SortedDict()
+        self.use_hot = config.local_cpu
+
+        #TODO: remove hardcode
+        dst_device = "cuda"
+        self.storage_backends: Dict[str, StorageBackendInterface] =\
+            CreateStorageBackends(config, metadata, dst_device)
+        self.prefetch_tasks: Dict[CacheEngineKey, Future] = {}
+        self.put_tasks: Dict[str, Dict[CacheEngineKey, Tuple[Future,
+                                                             MemoryObj]]] = {}
+        self.callback_executor = ThreadPoolExecutor(max_workers=1)
```

Comment:
We can remove this executor as callbacks are non-blocking.

Reply:
Done